pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -198,4 +206,28 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 	return errorutils.NewAggregate(errs)
 }
 
+func (spc *realStatefulPodControl) updatePersistentVolumeClaims(set *apps.StatefulSet, pod *v1.Pod) error {
+	var errs []error
+	for _, updatedClaim := range getPersistentVolumeClaims(set, pod) {
+		currentClaim, err := spc.pvcLister.PersistentVolumeClaims(updatedClaim.Namespace).Get(updatedClaim.Name)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Failed to retrieve PVC %s: %s", currentClaim.Name, err))
+			continue
+		}
+
+		currentStorageRequest := currentClaim.Spec.Resources.Requests[v1.ResourceStorage]
+		updatedStorageRequest := updatedClaim.Spec.Resources.Requests[v1.ResourceStorage]
+		if updatedStorageRequest.Cmp(currentStorageRequest) != 0 {
```

Comment:
this should check for expansion only right?

Reply:
Only allowing an increase is enforced in validation. But yes will change to `u.cmp(c) > 0` in case  a state like that is reached.